### PR TITLE
[Merged by Bors] - feat(Mathlib/RingTheory/Ideal/QuotientOperations): First Isomorphism Theorem for algebras

### PR DESCRIPTION
**Pull request by AntoineChambert-Loir**
We add the First Isomorphism Theorem for algebras.
(only the surjective version was present in mathlib).
We also adjust the docstring on top of the file.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

File: Mathlib/RingTheory/Ideal/QuotientOperations.lean [hunk 1]
```diff
@@ -644,6 +649,14 @@ lemma comap_map_mk {I J : Ideal R} (h : I ≤ J) :
     Ideal.comap (Ideal.Quotient.mk I) (Ideal.map (Ideal.Quotient.mk I) J) = J :=
   by ext; rw [← Ideal.mem_quotient_iff_mem h, Ideal.mem_comap]
 
+/-- The **first isomorphism theorem** for commutative algebras (`AlgHom.range` version). -/
+noncomputable def quotientKerEquivRange
+  {A B : Type*} [CommRing A] [Algebra R A] [Semiring B] [Algebra R B]
+  (f : A →ₐ[R] B) :
+  (A ⧸ RingHom.ker f) ≃ₐ[R] f.range :=
```
Comment (alreadydone) [inline Mathlib/RingTheory/Ideal/QuotientOperations.lean:656]: The linter should complain about the spacing ...

File: Mathlib/Algebra/Algebra/Subalgebra/Basic.lean [hunk 2]
```diff
@@ -626,6 +626,16 @@ def rangeRestrict (f : A →ₐ[R] B) : A →ₐ[R] f.range :=
   f.codRestrict f.range f.mem_range_self
 #align alg_hom.range_restrict AlgHom.rangeRestrict
 
+theorem rangeRestrict_surjective (f : A →ₐ[R] B):
+    Function.Surjective (f.rangeRestrict) :=
+  fun ⟨_y, hy⟩ =>
+  let ⟨x, hx⟩ := hy
+  ⟨x, SetCoe.ext hx⟩
```
Comment (alreadydone) [inline Mathlib/Algebra/Algebra/Subalgebra/Basic.lean:633]:
```suggestion
  RingHom.rangeRestrict_surjective f
```
(untested)

File: Mathlib/Algebra/Algebra/Subalgebra/Basic.lean [hunk 3]
```diff
@@ -626,6 +626,16 @@ def rangeRestrict (f : A →ₐ[R] B) : A →ₐ[R] f.range :=
...
+
+theorem ker_rangeRestrict (f : A →ₐ[R] B) :
+    RingHom.ker f.rangeRestrict = RingHom.ker f :=
+  Ideal.ext fun _ ↦ Subtype.ext_iff
```
Comment (alreadydone) [inline Mathlib/Algebra/Algebra/Subalgebra/Basic.lean:637]:
```suggestion
  rfl
```
(untested)

**Pull request closed without merging**